pkg/controller/mizar/grpc-adaptor.go

```diff
@@ -0,0 +1,272 @@
+		return getReturnCodeFromError(&err)
+	}
+	defer conn.Close()
+	defer cancel()
+	returnCode, err := client.UpdatePod(ctx, ConvertToPodContract(pod))
+	if err != nil {
+		return getReturnCodeFromError(&err)
+	}
+	return returnCode
+}
+
+// GrpcCreatePod is to invoking grpc func of CreatePod
+func GrpcCreatePod(grpcHost string, pod *v1.Pod) *ReturnCode {
+	client, ctx, conn, cancel, err := getGrpcClient(grpcHost)
+	if err != nil {
+		return getReturnCodeFromError(&err)
+	}
+	defer conn.Close()
+	defer cancel()
+	returnCode, err := client.CreatePod(ctx, ConvertToPodContract(pod))
+	if err != nil {
+		return getReturnCodeFromError(&err)
+	}
+	return returnCode
+}
+
+// GrpcDeletePod is to invoking grpc func of DeletePod
+func GrpcDeletePod(grpcHost string, pod *v1.Pod) *ReturnCode {
+	client, ctx, conn, cancel, err := getGrpcClient(grpcHost)
+	if err != nil {
+		return getReturnCodeFromError(&err)
+	}
+	defer conn.Close()
+	defer cancel()
+	returnCode, err := client.DeletePod(ctx, ConvertToPodContract(pod))
+	if err != nil {
+		return getReturnCodeFromError(&err)
+	}
+	return returnCode
+}
+
+// GrpcCreateNode is to invoking grpc func of CreateNode
+func GrpcCreateNode(grpcHost string, node *v1.Node) *ReturnCode {
+	client, ctx, conn, cancel, err := getGrpcClient(grpcHost)
+	if err != nil {
+		return getReturnCodeFromError(&err)
+	}
+	defer conn.Close()
+	defer cancel()
+	returnCode, err := client.CreateNode(ctx, ConvertToNodeContract(node))
+	if err != nil {
+		return getReturnCodeFromError(&err)
+	}
+	return returnCode
+}
+
+// GrpcResumeNode is to invoking grpc func of ResumeNode
+func GrpcResumeNode(grpcHost string, node *v1.Node) *ReturnCode {
+	client, ctx, conn, cancel, err := getGrpcClient(grpcHost)
+	if err != nil {
+		return getReturnCodeFromError(&err)
+	}
+	defer conn.Close()
+	defer cancel()
+	returnCode, err := client.ResumeNode(ctx, ConvertToNodeContract(node))
+	if err != nil {
+		return getReturnCodeFromError(&err)
+	}
+	return returnCode
+}
+
+// GrpcUpdateNode is to invoking grpc func of UpdateNode
+func GrpcUpdateNode(grpcHost string, node *v1.Node) *ReturnCode {
+	client, ctx, conn, cancel, err := getGrpcClient(grpcHost)
+	if err != nil {
+		return getReturnCodeFromError(&err)
+	}
+	defer conn.Close()
+	defer cancel()
+	returnCode, err := client.UpdateNode(ctx, ConvertToNodeContract(node))
+	if err != nil {
+		return getReturnCodeFromError(&err)
+	}
+	return returnCode
+}
+
+// GrpcDeleteNode is to invoking grpc func of DeleteNode
+func GrpcDeleteNode(grpcHost string, node *v1.Node) *ReturnCode {
+	client, ctx, conn, cancel, err := getGrpcClient(grpcHost)
+	if err != nil {
+		return getReturnCodeFromError(&err)
+	}
+	defer conn.Close()
+	defer cancel()
+	returnCode, err := client.DeleteNode(ctx, ConvertToNodeContract(node))
+	if err != nil {
+		return getReturnCodeFromError(&err)
+	}
+	return returnCode
+}
+
+func getReturnCodeFromError(err *error) *ReturnCode {
+	return &ReturnCode{
+		Code:    CodeType_TEMP_ERROR,
```

Comment:
I think HTTP Internal Error is more suitable in this case, which seems the usual behavior in K8s.

Reply:
One thing more:  getReturnCodeFromError and getGrpcClient are very likely to be used a lot by other team members. Consider change them to public functions